src/gt4py/next/iterator/tracing.py

```diff
@@ -210,6 +210,7 @@ def __bool__(self):
 class TracerContext:
```

Comment:
Why does the TracerContext need to support both closures and statements? Because the tests are written in the old way and we can change them until we updated the passes? Let's add a short todo with explanation here.

Reply:
added a comment